synda/model/node.py

```diff
@@ -57,12 +58,19 @@ def get(
         return results[0] if single_result and results else results
 
     @staticmethod
-    def get_from_step(session: Session, step: "Step") -> list["Node"]:
-        return session.exec(
+    def get_input_nodes_from_step(session: Session, step: "Step") -> list[tuple["Node", str]]:
+        input_nodes_for_steps = session.exec(
             select(Node)
             .join(StepNode, Node.id == StepNode.node_id)
-            .where(StepNode.step_id == step.id)
+            .where(and_(StepNode.step_id == step.id, StepNode.relationship_type == "input"))
         ).fetchall()
+        already_treated_input_nodes_ids = session.exec(
+            select(Node.parent_node_id).where(Node.parent_node_id.in_([node.id for node in input_nodes_for_steps]))
+        ).fetchall()
+        return [
+            (node, "treated") if node.id in already_treated_input_nodes_ids else (node, "not_treated")
+            for node in input_nodes_for_steps
+        ]
```

Comment:
Add a column `status` type ENUM with two values : `pending` or `processed` to make a more elegant request